elements/pl-file-editor/pl-file-editor.js

```diff
@@ -100,6 +108,56 @@ window.PLFileEditor.prototype.updatePreview = function (html_contents) {
   }
 };
 
+window.PLFileEditor.prototype.initSettingsButton = function (uuid) {
+  var that = this;
+
+  this.settingsButton.click(function () {
+    ace.require(['ace/ext/themelist'], function (themeList) {
+      var themeSelect = that.modal.find('#' + uuid + '-themes');
+      themeSelect.empty();
+      for (const entries in themeList.themesByName) {
+        var caption = themeList.themesByName[entries].caption;
+        var theme = themeList.themesByName[entries].theme;
+
+        if (localStorage.getItem('pl-file-editor-theme') === theme) {
+          themeSelect.append(
+            $('<option>', {
+              value: theme,
+              text: caption,
+              selected: true,
+            })
+          );
+        } else {
+          themeSelect.append(
+            $('<option>', {
+              value: theme,
+              text: caption,
+            })
+          );
+        }
+      }
+    });
+    that.modal.modal('show');
+    // listen for changes to the theme select
+    that.modal.find('#' + uuid + '-themes').change(function () {
+      var theme = $(this).val();
+      that.editor.setTheme(theme);
+      console.log('theme', theme);
+    });
+  });
+
+  this.saveSettingsButton.click(function () {
+    var theme = that.modal.find('#' + uuid + '-themes').val();
+    that.editor.setTheme(theme);
+    localStorage.setItem('pl-file-editor-theme', theme);
```

Comment:
Something to consider: if a question has multiple file editors, we should try to sync the settings between them so that the user doesn't have to reload the page for the change to take effect. It'd also be nice to have this sync immediately across tabs. For the latter, there's a [`storage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event) event that can be listened to. For the former... I'm not totally sure. Maybe we can use custom events for this?